src/java/org/apache/cassandra/cql3/statements/schema/AlterKeyspaceStatement.java

```diff
@@ -128,14 +142,21 @@ private void validateNoRangeMovements()
         if (allow_alter_rf_during_range_movement)
             return;
 
-        Stream<InetAddressAndPort> unreachableNotAdministrativelyInactive =
-            Gossiper.instance.getUnreachableMembers().stream().filter(endpoint -> !FBUtilities.getBroadcastAddressAndPort().equals(endpoint) &&
-                                                                      !Gossiper.instance.isAdministrativelyInactiveState(endpoint));
-        Stream<InetAddressAndPort> endpoints = Stream.concat(Gossiper.instance.getLiveMembers().stream(),
-                                                             unreachableNotAdministrativelyInactive);
-        List<InetAddressAndPort> notNormalEndpoints = endpoints.filter(endpoint -> !FBUtilities.getBroadcastAddressAndPort().equals(endpoint) &&
-                                                                                   !Gossiper.instance.getEndpointStateForEndpoint(endpoint).isNormalState())
-                                                               .collect(Collectors.toList());
+        ClusterMetadata metadata = ClusterMetadata.current();
```

Comment:
Referring `ClusterMetadata.current()` from the transformation code is wrong, so as referring to the configuration, `DatabaseDescriptor`, `Gossiper` and others. This can be seen in many schema transformations and makes the transformations non-pure. In turn, it can make replaying the log lead to inconsistent metadata for the same epoch on different nodes.